controller_manager:
  ros__parameters:
    update_rate: 5 
    use_sim_time: true
    drive_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    joint_broad:
      type: joint_state_broadcaster/JointStateBroadcaster

drive_controller:
  ros__parameters:
    use_sim_time: true
    joints:
      - FL_Steer2Servo
      - FR_Steer2Servo
      - BL_Steer2Servo
      - BR_Steer2Servo
      - fl_wheel
      - fr_wheel
      - bl_wheel
      - br_wheel
    command_interfaces:
      - position
      - velocity
    state_interfaces:
      - position
      - velocity
    gains:
      FL_Steer2Servo: {p: 0.001, d: 5, i: 0.1}
      FR_Steer2Servo: {p: 0.001, d: 5, i: 0.1}
      BL_Steer2Servo: {p: 0.001, d: 5, i: 0.1}
      BR_Steer2Servo: {p: 0.001, d: 5, i: 0.1}
      fl_wheel: {p: 0.001, d: 5, i: 0.1}
      fr_wheel: {p: 0.001, d: 5, i: 0.1}
      bl_wheel: {p: 0.001, d: 5, i: 0.1}
      br_wheel: {p: 0.001, d: 5, i: 0.1}